### PR TITLE
Add support for SSL access to the Pbench Server

### PR DESCRIPTION
**Pull request by webbnh**
This PR contains a sequence of enhancements relative to adding SSL support to the canned Pbench Server:
- Add a CA certificate and private key to be used to generate an SSL certificate for use by the Nginx proxy inside the container.  (Add the key's path to the `.gitleaks.toml` file, so that ITSec doesn't trigger on it when it is committed to a Git repo.)
- Enhance the `run-pbench-in-a-can` script to generate a certificate and private key and add them to the deployment files for the Pbench Server container.
- Modify the `deploy` script to map the SSL certificate and key into the Pbench Server container.
- Modify the configuration of the Nginx service in the container to listen for SSL connections on port `8443` and to use the mapped certificate and key.

Clients will be able to use HTTPS to access the Pbench Server, via the Nginx proxy.  The connection will not be trusted, so the user will either have to add our private CA's certificate to their trusted store or disable the security check in order to connect.

PBENCH-1149

### CODE REVIEW

File: server/pbenchinacan/run-pbench-in-a-can [hunk 1]
```diff
@@ -1,4 +1,5 @@
 #!/bin/bash -e
+# shellcheck disable=SC2086,2174
```
Comment (ndokos) [inline server/pbenchinacan/run-pbench-in-a-can:2]: Doesn't the second one need a prefix too?

**Comment by webbnh** [comment]
> Looks plausible: I guess the real test is whether we can stand up a `runlocal` and do a `GET https://localhost:8080/api/v1/endpoints`.

What I did was run `jenkins/run-server-func-tests` without the  `--cleanup` and then hit `https://10.1.169.200:8443/api/v1/endpoints` and `https://10.1.169.200:8443/dashboard` with my browser (after importing the CA to my "login keychain" as a trusted root CA).

> A functional test case would be nice... but since this apparently doesn't and isn't likely to break anything that works now, I'll approve.

My intention/assumption is that _this_ PR only **adds** HTTPS access (and, therefore, as long as no existing functional test fails that proves that it "works") and that the "next" PR will remove HTTP access...along with converting the functional tests to use HTTPS...and, therefore, no new functional tests will be required.  (So, I don't see the point of adding a functional test which will quickly end up being redundant.)

**Review by npalaska — APPROVED** [review]
This looks good and may get us one (or multiple) steps closer to getting the IT-supported SSO.

**Review by ndokos — APPROVED** [review]
LGTM but that does not mean much. I had one nit-question that I will pre-resolve.

**Comment by dbutenhof** [comment]
> My intention/assumption is that _this_ PR only **adds** HTTPS access (and, therefore, as long as no existing functional test fails that proves that it "works")

Technically, the current state of the CI only proves that you didn't break `http`, not that `https` works. You're right that we'll eventually withdraw `http` and rely entirely on `https`; in the meantime we don't have any continuing proof that you've moved us a step closer to that goal.

I did approve, which means I'm willing to accept that gap in the hope that it'll be short and the code won't have time to rot ... but I'd feel better with enduring proof. 😀

**Comment by webbnh** [comment]
> Technically, the current state of the CI only proves that you didn't break `http`, not that `https` works.

True.  I'm taking baby steps and trying hard not to break anything in the process.  😇 

> in the meantime we don't have any continuing proof that you've moved us a step closer to that goal.

True...and, I'm asserting that that doesn't matter (much)...except to the extent that the "next" PR will be much quicker and easier than it otherwise would have been.

> it'll be short and the code won't have time to rot ... but I'd feel better with enduring proof. 😀

I feel your concern at least as deeply as you do...but I don't think the proof will matter -- we'll either use this stuff or we won't, and, in the event that we don't, I don't think it will have a chance to rot....

**Pull request merged**